internal/license/main.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	// how many lines to check for an existing copyright
+	// this logic is not great and we should probably do something else
+	// but this was copied from the python script
+	copyrightLineLimit = 5
+	headerPrefix       = "// Copyright (c)"
+)
+
+var (
+	headerString  = fmt.Sprintf(headerPrefix+" %d Uber Technologies, Inc.", time.Now().UTC().Year())
```

Comment:
Ya I can just do that, I would want to clean up the code a bit. Optimally it would be nice to add features so it's not so uber-specific or golang-specific (ie default to MIT license, default to "Uber Technologies, Inc.", default to the current year on your machine, support multiple extensions, default to all go files not in vendor).